src/hooks/useFetchExperience.ts

```diff
@@ -0,0 +1,194 @@
+import type { ContentfulClientApi } from 'contentful'
+import { useCallback, useState } from 'react'
+import { fetchExperienceEntities, fetchExperienceEntry } from '../core/fetchers'
+import { Experience, ExternalSDKMode } from '../types'
+import { createExperience } from '../core'
+
+const errorMessagesWhileFetching = {
+  experience: 'Failed to fetch experience',
+  experienceReferences: 'Failed to fetch entities, referenced in experience',
+}
+
+const handleError = (generalMessage: string, error: unknown) => {
+  const message = error instanceof Error ? error.message : `Unknown error: ${error}`
+  console.error(`${generalMessage} with error: ${message}`)
+}
+
+type useClientsideExperienceFetchersProps = {
+  mode: ExternalSDKMode
+  client: ContentfulClientApi<undefined>
+}
+
+export const useFetchExperience = ({ mode, client }: useClientsideExperienceFetchersProps) => {
+  const [experience, setExperience] = useState<Experience | undefined>(undefined)
+  const [isFetching, setIsFetching] = useState(false)
+
+  /**
+   * Fetch experience entry using slug as the identifier
+   * @param {string} experienceTypeId - id of the content type associated with the experience
+   * @param {string} slug - slug of the experience (defined in entry settings)
+   * @param {string} localeCode - locale code to fetch the experience. Falls back to the currently active locale in the state
+   */
+  const fetchBySlug = useCallback(
+    async ({
+      experienceTypeId,
+      slug,
+      localeCode,
+    }: {
+      experienceTypeId: string
+      slug: string
+      localeCode: string
+    }): Promise<{ success: boolean; error?: Error; experience?: Experience }> => {
```

Comment:
Promises are built for having a `resolve` and `reject` case. What is the advantage of having a custom success and error attribute vs using the native approach (which an un-experience developer probably might expect).

Reply:
Good call. The shape of the response has been inherited from previous implementation. I think we can change it actually

Reply:
Applied